### ISSUE TITLE
Redundant errors in html-checker

### ISSUE THREAD

**Issue by qzhou1607** [sonarwhal/sonar#474]
connector: `chrome` and `jsdom`
url: www.gmail.com
rule: html-checker

About 366 errors were reported from `html-checker`.
![html-checker](https://user-images.githubusercontent.com/20218531/29898602-67d36fc0-8d9b-11e7-8236-95335844962b.PNG)

But running `https://www.gmail.com` in https://validator.w3.org/nu/ returns only 4 errors. For this particular page, the html returned from our connector seems to be different from the content fetched directly by the validator using the url.

**Comment by molant** [comment]
> But running https://www.gmail.com in https://validator.w3.org/nu/ returns only 4 errors. For this particular page, the html returned from our connector seems to be different from the content fetched directly by the validator using the url.

I believe this is because the w3c validator only fetches the url and that's all, while the connectors execute some JS on that page and then send the HTML to the validator. You have the option to check the source the validator is analyzing, if you compare it to the source of the page once it is loaded you will see it is different so IMO, the results of sonar are more accurate.

-----

In any case, maybe we should do some changes to the validator rule as it can generate a lot of noise.
Some ideas (not exclusive):

* Not be a default one
* Report errors grouped if possible
* Just report one error and link to the results page

@sonarwhal/contributors what do you think?

**Comment by qzhou1607** [comment]
@molant What do you think about the following? 
 * By default we group the errors by error messages, so users have a rough idea about what `html-checker` complains about.
* The user is allowed to enable showing the details, so if they are interested in the line/column numbers and etc for the `html-checker` errors, they can configure to do so.

**Comment by molant** [comment]
> The user is allowed to enable showing the details, so if they are interested in the line/column numbers 

so we will have an option for this rule? Something like the following?

```json
"html-checker": ["error", {
  "report": "compact|detailed"
}]
```

I like that idea. We need to see how the grouping would be like because sometimes is very specific ("you can't use attribute ATTRIBUTE in here". Will be nice if they had an error id or similar.

Also I still have mixed feelings about having it on by default.

**Comment by qzhou1607** [comment]
Yes. 
> so we will have an option for this rule? Something like the following?

It doesn't provide error id, but returns the location information and code snippet for each error. I don't think they include location information inside the error messages, but sometimes it will refer to the location like `Attribute XX is not allowed at this point/in this context`.
> Will be nice if they had an error id or similar.
